tests/test_cli.py

```diff
@@ -0,0 +1,114 @@
+"""
+Tests the cli
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+import argparse
+
+import tests
+import ga4gh.cli as cli
+import ga4gh.protocol as protocol
+import ga4gh.client as client
+
+
+class TestNoInput(unittest.TestCase):
+    """
+    Test the cli as if there was no input; print_help should be called
+    """
+    def testServerNoInput(self):
+        parser = StubArgumentParser()
+        cli.server_main(parser)
+        parser.assertCounts(1, 1)
+
+    def testCliNoInput(self):
+        parser = StubArgumentParser()
+        cli.client_main(parser)
+        parser.assertCounts(1, 1)
+
+
+class TestClientArguments(unittest.TestCase):
+    """
+    Tests the client cli can parse all arguments it is supposed to
+    and can initialize the runner in preparation for a request
+    """
+    def setUp(self):
+        # initialize the client parser
+        self.parser = StubArgumentParser()
+        cli.client_main(self.parser)
+
+    def tearDown(self):
```

Comment:
`tearDown` shouldn't include any assertions, as failures here will (I think) be treated as an error not a test failure. I think a better structure would be 
```python
def verifyInput(self, inputString):
     """
     [docstring comment]
     """
     # put in all the stuff you have in tearDown now

def testVariantsSearchArguments(self):
     inputStr = """..."""
     self.verifyInput(inputStr)
   
    # etc
```